pkg/storageops/vsphere/vsphere_util.go

```diff
@@ -0,0 +1,157 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/libopenstorage/openstorage/pkg/storageops"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere/vclib"
+)
+
+// derived from https://github.com/kubernetes/kubernetes/blob/release-1.9/pkg/cloudprovider/providers/vsphere/vsphere_util.go#L94
+// VSphereConfig represents the vsphere configuration
+type VSphereConfig struct {
+	// User is the vCenter username.
+	User string
+	// Password is the vCenter password in clear text.
+	Password string
+	// VCenterIP is the vcenter IP to connect on
+	VCenterIP string
+	// VCenterPort is the vcenter port to connect on
+	VCenterPort string
+	// InsecureFlag True if vCenter uses self-signed cert.
+	InsecureFlag bool
+	// Datastore where all disks should be created
+	Datastore string
+	// RoundTripperCount is the Soap round tripper count (retries = RoundTripper - 1)
+	RoundTripperCount uint
+	// VMUUID is the VM Instance UUID of virtual machine which can be retrieved from instanceUuid
+	// property in VmConfigInfo, or also set as vc.uuid in VMX file.
+	// If not set, will be fetched from the machine via sysfs (requires root)
+	VMUUID string
+}
+
+// Represents a vSphere instance where one or more kubernetes nodes are running.
+type VSphereInstance struct {
+	conn *vclib.VSphereConnection
+}
+
+var datastoreFolderIDMap = make(map[string]map[string]string)
+
+// Structure that represents Virtual Center configuration
+type VirtualCenterConfig struct {
+	// vCenter username.
+	User string
+	// vCenter password in clear text.
+	Password string
+	// vCenter port
+	VCenterPort string
+	// Datacenter in which VMs are located.
+	Datacenter string
+	// Soap round tripper count (retries = RoundTripper - 1)
+	RoundTripperCount uint
+}
+
+func ReadVSphereConfigFromEnv() (*VSphereConfig, error) {
+	var cfg VSphereConfig
+	var err error
+
+	cfg.VCenterIP, err = storageops.GetEnvValueStrict("VSPHERE_VCENTER")
+	if err != nil {
+		return nil, err
+	}
+	cfg.VCenterPort, err = storageops.GetEnvValueStrict("VSPHERE_VCENTER_PORT")
+	if err != nil {
+		return nil, err
+	}
+	cfg.User, err = storageops.GetEnvValueStrict("VSPHERE_USER")
+	if err != nil {
+		return nil, err
+	}
+	cfg.Password, err = storageops.GetEnvValueStrict("VSPHERE_PASSWORD")
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.Datastore, _ = storageops.GetEnvValueStrict("VSPHERE_DATASTORE")
+
+	cfg.InsecureFlag = false
+	insecure, err := storageops.GetEnvValueStrict("VSPHERE_INSECURE")
+	if err == nil && strings.ToLower(insecure) == "true" {
+		cfg.InsecureFlag = true
+	}
+	return &cfg, nil
+}
+
+// IsDevMode checks if requirement env variables are set to run the pkg outside vsphere in dev mode
+func IsDevMode() bool {
+	_, err := storageops.GetEnvValueStrict("VSPHERE_VM_UUID")
```

Comment:
Why is VMSPHERE_VM_UUID not a part of the above ReadVSphereConfigFromEnv function?

Reply:
added it back. It was needed